### PR TITLE
Enable ruff's flake8-implicit-str-concat (ISC) rules

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Enable [ISC](https://docs.astral.sh/ruff/rules/#flake8-implicit-str-concat-isc) rules.

Address https://github.com/GenericMappingTools/pygmt/issues/2741.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -768,8 +768,7 @@ def _parse_constant(self, constant, valid, valid_modifiers=None):
             )
         if nmodifiers > 0 and valid_modifiers is None:
             raise GMTInvalidInput(
-                "Constant modifiers not allowed since valid values were not "
-                + f"given: '{constant}'"
+                "Constant modifiers not allowed since valid values were not given: '{constant}'"
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:771]:
```suggestion
                "Constant modifiers are not allowed since valid values were not given: '{constant}'"
```

**Review by yvonnefroehlich — COMMENTED** [review]
Sofar I did not follow the PRs related to issue #2741 in greater detail. So, maybe I am missing some context here, but I am wondering why the `ISC` rule is not added to the list in `pyproject.toml` in contrast to the other `ruff` rules:
https://github.com/GenericMappingTools/pygmt/blob/78dfcf13718f56a755cc16b03ac29029f26235ec/pyproject.toml#L100-L112

**Comment by seisman** [comment]
> Sofar I did not follow the PRs related to issue #2741 in greater detail. So, maybe I am missing some context here, but I am wondering why the `ISC` rule is not added to the list in `pyproject.toml` in contrast to the other `ruff` rules:
> 
> https://github.com/GenericMappingTools/pygmt/blob/78dfcf13718f56a755cc16b03ac29029f26235ec/pyproject.toml#L100-L112

Thanks for catching it. I forgot to add it. Now added in 0d78ee5aa569b8b8797620e0ff758ee67f550814

File: pyproject.toml [hunk 2]
```diff
@@ -113,6 +114,7 @@ select = [
 ]
 ignore = [
     "E501", # Avoid enforcing line-length violations
+    "ISC001", # single-line-implicit-string-concatenation, conflict with formatter
```
Comment (michaelgrund) [inline pyproject.toml:117]:
```suggestion
    "ISC001", # Single-line-implicit-string-concatenation, conflict with formatter
```

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**